capa/render/result_document.py

```diff
@@ -418,8 +418,9 @@ def from_capa(
                                     and a.id <= location.id
                                 ]
                             )
-                            _, most_recent_match = matches_in_thread[-1]
-                            children.append(Match.from_capa(rules, capabilities, most_recent_match))
+                            if matches_in_thread:
```

Comment:
would you add a comment here reminding us why this might not be the case? otherwise, looks good!